pkg/debug/api/cell.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package api
+
+import (
+	"github.com/cilium/hive/cell"
+	"github.com/spf13/cast"
+
+	restapi "github.com/cilium/cilium/api/v1/server/restapi/daemon"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/loadbalancer/legacy/service"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/status"
+	wireguard "github.com/cilium/cilium/pkg/wireguard/agent"
+)
+
+// Cell provides the Debug-Info API handler
+var Cell = cell.Module(
+	"debuginfo-api",
+	"Debuginfo API handler",
+
+	cell.Provide(newDebugAPIHandler),
+
+	// Provide a read-only copy of the current daemon settings to be consumed
+	// by the debuginfo API
+	cell.ProvidePrivate(daemonSettings),
+)
+
+type debugAPIHandlerParams struct {
+	cell.In
+
+	EndpointManager endpointmanager.EndpointManager
+	PolicyRepo      policy.PolicyRepository
+	ServiceManager  service.ServiceManager
+	WireguardAgent  *wireguard.Agent
+
+	StatusCollector status.StatusCollector
+	CellSettings    cellSettings
+}
+
+type debugAPIHandlerOut struct {
+	cell.Out
+
+	GetDebuginfoHandler restapi.GetDebuginfoHandler
+}
+
+func newDebugAPIHandler(params debugAPIHandlerParams) debugAPIHandlerOut {
```

Comment:
nit: out struct just as a matter of principle or could this just be `GetDebuginfoHandler`?

Reply:
Yes, it would also be possible to just return `GetDebuginfoHandler` in this case. I thought it's best to keep it the same as I already did it for other modules (where a module might provide multiple handlers)